packages/react-data-grid/src/__tests__/Cell.spec.tsx

```diff
@@ -0,0 +1,168 @@
+import React, { PropsWithChildren } from 'react';
+import { shallow } from 'enzyme';
+
+import Cell, { Props } from '../Cell';
+import helpers from '../helpers/test/GridPropHelpers';
+import CellAction from '../CellAction';
+import { SimpleCellFormatter } from '../formatters';
+import { Column, CellMetaData } from '../common/types';
+
+const testCellMetaData: CellMetaData = {
+  rowKey: 'row',
+  onCellClick() {},
+  onCellContextMenu() {},
+  onCellDoubleClick() {},
+  onDragEnter() {},
+  onCellExpand() {},
+  onRowExpandToggle() {}
+};
+
+const expandableOptions = {
+  canExpand: false,
+  field: 'field',
+  expanded: false,
+  children: [],
+  treeDepth: 0,
+  subRowDetails: {
+    canExpand: false,
+    field: 'field',
+    expanded: false,
+    children: [],
+    treeDepth: 0,
+    siblingIndex: 0,
+    numberSiblings: 0
+  }
+};
+
+const defaultColumn: Column = { key: 'col1', name: 'col1', width: 100, left: 0 };
+
+const testProps: Props = {
+  rowIdx: 0,
+  idx: 1,
+  column: defaultColumn,
+  value: 'Wicklow',
+  cellMetaData: testCellMetaData,
+  rowData: { name: 'Johnny Test', location: 'Wicklow', likesTesting: 'Absolutely' },
+  height: 40,
+  isScrolling: false,
+  scrollLeft: 0
+};
+
+const renderComponent = (extraProps?: PropsWithChildren<Partial<Props>>) => {
+  return shallow<Cell>(<Cell {...testProps} {...extraProps} />);
+};
+
+describe('Cell Tests', () => {
+  it('should render a SimpleCellFormatter with value', () => {
+    const wrapper = renderComponent();
+    const formatter = wrapper.find(SimpleCellFormatter);
+    expect(formatter.props().value).toEqual('Wicklow');
+  });
+
+  it('should render a custom formatter when specified on column', () => {
+    const CustomFormatter = () => <div>Custom</div>;
+
+    const column = {
+      ...defaultColumn,
+      formatter: CustomFormatter
+    };
+
+    const wrapper = renderComponent({ column });
+    const formatterInstance = wrapper.find(CustomFormatter);
+    expect(formatterInstance.prop('value')).toEqual('Wicklow');
+  });
+
+  it('should render children when those are passed', () => {
+    const wrapper = renderComponent({ children: <div>Child</div> });
+    expect(wrapper.children().text()).toBe('Child');
+  });
+
+  describe('Rendering Cell component', () => {
+    function shallowRenderComponent(props: Props) {
+      return shallow<Cell>(<Cell {...props} />);
+    }
+
+    const requiredProperties: Props = {
+      rowIdx: 18,
+      idx: 19,
+      height: 60,
+      column: helpers.columns[0],
+      value: 'requiredValue',
+      cellMetaData: testCellMetaData,
+      rowData: helpers.rowGetter(11),
+      expandableOptions,
+      isScrolling: false,
+      scrollLeft: 0
+    };
+
+    it('passes classname property', () => {
+      const wrapper = shallowRenderComponent(requiredProperties);
+      const cellDiv = wrapper.find('div').at(0);
+      expect(cellDiv.hasClass('react-grid-Cell'));
```

Comment:
for the test at ln#98, ln#103 and ln#108, I believe they can all be caught by 1 snapshot testing.

Reply:
I guess so, @amanmahajan7 do we want to add snapshot testing to RDG?